packages/common/src/compiler/filtersCompiler.ts

```diff
@@ -107,111 +108,130 @@ export const renderNumberFilterSql = (
     }
 };
 
+// from 0 (Monday) to 6 (Sunday) to 0 (Sunday) to 6 (Saturday)
+const convertWeekDayToMomentWeekDay = (weekDay: WeekDay) => {
+    const converted = weekDay + 1;
+    return converted <= 6 ? converted : 0;
+};
 export const renderDateFilterSql = (
     dimensionSql: string,
     filter: DateFilterRule,
     dateFormatter: (date: Date) => string = formatDate,
+    startOfWeek: WeekDay | null | undefined = undefined,
 ): string => {
     const filterType = filter.operator;
-    switch (filter.operator) {
-        case 'equals':
-            return `(${dimensionSql}) = ('${dateFormatter(
-                filter.values?.[0],
-            )}')`;
-        case 'notEquals':
-            return `(${dimensionSql}) != ('${dateFormatter(
-                filter.values?.[0],
-            )}')`;
-        case 'isNull':
-            return `(${dimensionSql}) IS NULL`;
-        case 'notNull':
-            return `(${dimensionSql}) IS NOT NULL`;
-        case 'greaterThan':
-            return `(${dimensionSql}) > ('${dateFormatter(
-                filter.values?.[0],
-            )}')`;
-        case 'greaterThanOrEqual':
-            return `(${dimensionSql}) >= ('${dateFormatter(
-                filter.values?.[0],
-            )}')`;
-        case 'lessThan':
-            return `(${dimensionSql}) < ('${dateFormatter(
-                filter.values?.[0],
-            )}')`;
-        case 'lessThanOrEqual':
-            return `(${dimensionSql}) <= ('${dateFormatter(
-                filter.values?.[0],
-            )}')`;
-        case FilterOperator.IN_THE_PAST: {
-            const unitOfTime: UnitOfTime =
-                filter.settings?.unitOfTime || UnitOfTime.days;
-            const completed: boolean = !!filter.settings?.completed;
+    const currentLocale = moment.locale();
+    if (isWeekDay(startOfWeek)) {
+        moment.locale('lightdash-custom', {
+            week: {
+                dow: convertWeekDayToMomentWeekDay(startOfWeek),
+            },
+        });
+        moment.locale('lightdash-custom');
```

Comment:
I believe this first call is the declaration and the second one is an instantiation.
but also, I think it just overrides the global locale for the imported moment instance. is there a way to instantiate moment first and then change locale for that particular instance, since this is being used by the backend process?

also, I feel like this is a chunk of code that can live somewhere in utils